src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs

```diff
@@ -102,6 +102,47 @@ public void EditSimulation(ISimulation simulation)
          Refresh();
       }
 
+
+      private void removeFromOutputMappingList(IReadOnlyList<DataRepository> itemsToRemove)
+      {
+         var idsToRemove = new HashSet<string>(
+             (itemsToRemove ?? Enumerable.Empty<DataRepository>())
+             .Where(item => item != null)
+             .Select(item => item.Id)
+         );
+
+         for (int i = _listOfOutputMappingDTOs.Count - 1; i >= 0; i--)
+         {
+            var dto = _listOfOutputMappingDTOs[i];
+            var observedData = dto.ObservedData;
+
+            if (observedData == null)
+               continue;
+
+            if (idsToRemove.Contains(observedData.Id))
+               _listOfOutputMappingDTOs.RemoveAt(i);
+         }
+
+         var mappedIds = new HashSet<string>(
+             _listOfOutputMappingDTOs
+                 .Where(dto => dto.ObservedData != null)
+                 .Select(dto => dto.ObservedData.Id)
+         );
+
+         foreach (var observedData in allAvailableObservedDataUsedBySimulation())
+         {
+            if (!mappedIds.Contains(observedData.Id))
+            {
+               var newOutputMapping = new OutputMapping
+               {
+                  WeightedObservedData = new WeightedObservedData(observedData)
+               };
+               var newOutputMappingDTO = mapFrom(newOutputMapping);
+               _listOfOutputMappingDTOs.Add(newOutputMappingDTO);
+            }
+         }
```

Comment:
I think you only need 
```
_listOfOutputMappingDTOs.Where(x => itemsToRemove.Contains(x.ObservedData)).ToList().Each(x => _listOfOutputMappingDTOs.Remove(x));
```

It should not be necessary to add anything into the list after you remove the `itemsToRemove`